src/checkpoint.rs

```diff
@@ -0,0 +1,370 @@
+//! This module supports saving and restoring variables using Tensorflow checkpoints in SaveV2 format
+
+use crate::option_insert_result::OptionInsertWithResult;
+use crate::{ops, Operation, Scope, Session, SessionRunArgs, Status, Tensor};
+
+#[derive(Debug)]
+struct SaveRestoreOps {
+    pub prefix_save: Operation,
+    pub prefix_restore: Operation,
+    pub save_op: Operation,
+    pub restore_op: Operation,
+}
+
+/// Checkpointing and restoring struct
```

Comment:
This documentation needs to be a little more in-depth.

Reply:
Ok, I added documentation and a small example at the start of the module.